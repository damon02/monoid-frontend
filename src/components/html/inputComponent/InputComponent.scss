@import '../../../style/style.scss';

.inputComponent {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  width: 100%;
  height: 100%;
  max-width: 320px;
  margin-top: 12px;
  margin-bottom: 12px;

  .abs-fa {
    position: absolute;
    top: calc(50% - 8px);
    left: $spaces-medium;
    color: $color-dark-gray;
  }

  .label {
    position: absolute;
    left: 7px;
    top: -7px;
    font-family: 'Open Sans';
    font-weight: 200;
    text-transform: uppercase; 
    font-size: 12px;
    background-color: #fff;
    padding: 0px 4px;
    color: $color-dark-gray; 
    z-index: 2;
    &.empty {
      top: calc(50% - 10px); 
      left: 7px;
      font-size: 14px;
      background-color: unset;
      z-index: -1;
    }
    
    -webkit-transition: all .3s ease;
    -moz-transition: all .3s ease;
    -ms-transition: all .3s ease;
    -o-transition: all .3s ease;
    transition: all .3s ease;
  }
  
  input {
    height: 100%;
    min-height: 52px;
    padding-left: 10px;
    border: none;
    background-color: unset;
    border: 1px solid $color-gray;
    border-bottom: 2px solid $color-gray;
    z-index: 1;
    font-size: 16px;
    font-family: 'Proxima Nova';
    &.untouched {
      border-bottom: 1px solid $color-gray;
    }
    &:focus {
      border-bottom: 2px solid $color-black;
      outline: 0;
    }
    .invalid & {
      border-bottom: 2px solid $color-red;
    }
    .valid & {
      border-bottom: 2px solid $color-green;
    }
  }

  &.icon-padding {
    input {
      padding-left: 34px;
    }
    .label {
      left: 32px;
    }
  }
}