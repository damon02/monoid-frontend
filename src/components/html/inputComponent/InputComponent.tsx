import * as React from 'react'
import './InputComponent.scss'

interface IInputComponentProps {
  label: string
  type: 'date' | 'email' | 'number' | 'password' | 'text' | 'url'
  value: string
  onChange: (value : string) => void
  isValid?: boolean
  onFocus?: (x: boolean) => void
  touched?: boolean
  autoComplete?: 'on' | 'off'
  faIcon?: string
}


const InputComponent = (props : IInputComponentProps) => (
  <div className={`inputComponent ${props.isValid !== undefined ? props.isValid === false ? ' invalid' : ' valid' : ''}${props.faIcon ? ' icon-padding' : ''}`}>
    <label className={`label ${props.value ? 'extended' : 'empty'}`}>{props.label}</label>
    {props.faIcon ? <i className={`abs-fa ${props.faIcon}`}/> : null}
    <input 
      className={`field${props.touched ? '' : ' untouched'}${props.faIcon ? ' icon-padding' : ''}`}
      type={props.type}
      value={props.value}
      onChange={(e) => !props.onChange ? ({}) : props.onChange(e.target.value)}
      onFocus={() => !props.onFocus ? ({}) : props.onFocus(true)}
      autoComplete={props.autoComplete || 'on'}
    />
  </div>
)

export default InputComponent