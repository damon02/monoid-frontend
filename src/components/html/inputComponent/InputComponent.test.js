import React from 'react'
import ReactDOM from 'react-dom'
import InputComponent from './InputComponent'
import Renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <InputComponent />, 
    div
  )
})

it('renders correctly', () => {
  const tree = Renderer
    .create(<InputComponent label={'test'} type={'text'} value={'hello world'} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('changes input and then renders correctly', () => {
  const input = shallow(<InputComponent label={'test'} type={'text'} value={''} />)

  expect(input.containsMatchingElement(<label>test</label>))
  expect(input.containsMatchingElement(<input></input>))

  input.setProps({ value: 'Hello world!' })

  expect(input.containsMatchingElement(<input>Hello world!</input>))
})