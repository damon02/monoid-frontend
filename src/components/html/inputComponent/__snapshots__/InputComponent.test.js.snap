// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="inputComponent "
>
  <label
    className="label extended"
  >
    test
  </label>
  <input
    autoComplete="on"
    className="field untouched"
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="hello world"
  />
</div>
`;
